json-rpc/src/methods.rs

```diff
@@ -393,6 +394,44 @@ async fn get_transactions(
     Ok(result)
 }
 
+/// Returns transactions by range with proofs
+async fn get_transactions_with_proofs(
+    service: JsonRpcService,
+    request: JsonRpcRequest,
+) -> Result<Option<TransactionsWithProofsView>> {
+    let start_version: u64 = request.parse_param(0, "start_version")?;
+    let limit: u64 = request.parse_param(1, "limit")?;
+
+    // Notice limit is a u16 normally, but some APIs require u64 below
+    service.validate_page_size_limit(limit as usize)?;
+
+    if start_version > request.version() {
+        return Ok(None);
+    }
+
+    // We do not fetch events since they don't come with proofs.
+    let txs = service
+        .db
+        .get_transactions(start_version, limit, request.version(), false)?;
+
+    let mut blobs = vec![];
+    for t in txs.transactions.iter() {
+        let bv = lcs::to_bytes(t)?;
+        blobs.push(BytesView::from(bv));
+    }
+
+    let (proofs, tx_info) = txs.proof.unpack();
+
+    Ok(Some(TransactionsWithProofsView {
+        serialized_transactions: blobs,
+        ledger_info: BytesView::from(&lcs::to_bytes(&request.ledger_info)?),
```

Comment:
We talked about this last time: we should consider returning ledger info with proof or just let client batch call with state_with_proof method.
I think in case we want batch call to get more txns with proof, or get account state with proof, maybe batch call is better.

Reply:
How can you do a batch call currently to be sure that both reqwest's ledger_info would be the same?

Reply:
Done